libraries/azurerm_storage_accounts.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class StorageAccounts < AzurermPluralResource
+  name 'azurerm_storage_accounts'
+  desc 'Fetches all Azure Monitor Log Profiles'
+  example <<-EXAMPLE
+    describe azurerm_storage_accounts do
+      its('names') { should include('default') }
+    end
+  EXAMPLE
+
+  FilterTable.create
+             .register_column(:names, field: 'name')
+             .install_filter_methods_on_resource(self, :table)
+
+  attr_reader :table
+
+  def initialize
+    resp = management.storage_accounts
+    return if has_error?(resp)
+
+    @table = resp
+  end
+
+  include Azure::Deprecations::StringsInWhereClause
```

Comment:
What does this do?

Reply:
This includes a deprecation warning from previous implementations that keyed plural resources by string keys e.g. `'MyKey'` instead of using symbols. 

By including this you'll monkey patch this resource to redefine the `where` method to check if you used a string instead of a symbol. If you did, then it will convert the string to a symbol and give a deprecation warning.

This should not be included as it's a new resource and you should not be doing any filtering on strings.

Reply:
An example of poor cargo culting.